packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,52 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    const attributeMap = {};
```

Comment:
This is not safe: what if there is an attribute named, for example, `hasOwnProperty`? In cases like this, we should use either `Object.create(null)` or `new Map()`.

Reply:
Also, maybe add a test where an attribute is called `hasOwnProperty`

Reply:
Added this test as well 👍